src/vic3_world/institutions/institutions_importer.cpp

```diff
@@ -0,0 +1,51 @@
+#include "institutions_importer.h"
+
+#include "src/vic3_world/database/database_parser.h"
+
+namespace vic3
+{
+
+InstitutionsImporter::InstitutionsImporter(std::map<int, std::vector<Institution>>& out_institutions_map):
+    institutions_map_(out_institutions_map),
+    current_institution_()
+{
+}
+
+void InstitutionsImporter::operator()(std::istream& input_stream)
+{
+   // create single institution parser
+   // this has to be in the operator() method so this object can be copied/moved and still work
+   CallableParser institution_parser_;
```

Comment:
You likely already noticed, but the tradeoff is that you're rebuilding the parser each time, and that adds up. In past converters that was a significant slowdown, though probably mostly from regexes, which you aren't using here.

Reply:
Yeah, due to the way the parser works, we need a stable address to parse the element contents to, so either (a) unique_ptr or (b) inline lambdas, and the rest of the code seems to use runtime lambdas. This code made me understand why java went with the "all objects are malloc'd" approach.